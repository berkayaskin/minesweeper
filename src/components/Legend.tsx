import { Box, BoxProps, HStack, Text, TextProps } from '@chakra-ui/react'

export const Legend = () => {
  return (
    <Parent>
      <Text as='h1' fontSize='lg' color='whiteAlpha.800' mr={1}>
        flag:{' '}
      </Text>
      <Key color='red.600'>ctrl</Key>
      <Text mx={1}>+</Text>
      <Key color='blue.600'>click</Key>
    </Parent>
  )
}

// Path: src/components/Key.tsx
const Parent = (props: BoxProps) => (
  <Box as='code' fontSize='sm'>
    <HStack justifyContent='center'>{props.children}</HStack>
  </Box>
)

// Path: src/components/Key.tsx
const Key = (props: TextProps) => (
  <Text
    as='span'
    display='inline-block'
    bgColor='gray.400'
    py={1}
    px={2}
    borderRadius={3}
    {...props}
  >
    {props.children}
  </Text>
)
