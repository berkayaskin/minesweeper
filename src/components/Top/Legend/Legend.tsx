import { Box, BoxProps, HStack, Text, TextProps } from '@chakra-ui/react'

export interface LegendProps {
  /**
   * Feature that should be activated after actions
   * @example 'Flag'
   * @example 'Question'
   * @example 'Reveal'
   * @example 'Restart'
   * @example 'New Game'
   */
  feature: string
  /**
   * First action that is required to activate feature
   * @example 'ctrl'
   * @example 'alt'
   * @example 'shift'
   */
  firstAction: string
  /**
   * Second action that is required to activate feature
   * @example 'click'
   * @example 'right click'
   * @example 'double click'
   * @example 'drag'
   * @example 'hover'
   */
  secondAction: string
}

export const Legend = ({ feature, firstAction, secondAction }: LegendProps) => {
  return (
    <Parent>
      <Text as='h1' fontSize='lg' color='whiteAlpha.800' mr={1}>
        {feature}:
      </Text>
      <FirstAction>{firstAction}</FirstAction>
      <Text mx={1}>+</Text>
      <SecondAction>{secondAction}</SecondAction>
    </Parent>
  )
}

const Parent = (props: BoxProps) => (
  <Box as='code' fontSize='sm'>
    <HStack justifyContent='center'>{props.children}</HStack>
  </Box>
)

const FirstAction = (props: TextProps) => (
  <Text
    as='span'
    color='red.600'
    display='inline-block'
    bgColor='gray.400'
    py={1}
    px={2}
    borderRadius={3}
    {...props}
  >
    {props.children}
  </Text>
)

const SecondAction = (props: TextProps) => (
  <Text
    as='span'
    color='blue.600'
    display='inline-block'
    bgColor='gray.400'
    py={1}
    px={2}
    borderRadius={3}
    {...props}
  >
    {props.children}
  </Text>
)
