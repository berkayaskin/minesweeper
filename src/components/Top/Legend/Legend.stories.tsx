import { Meta, Story } from '@storybook/react/types-6-0'
import { Legend } from './Legend'

export default {
  title: 'Top/Legend',
  component: Legend,
} as Meta

const Template: Story = (args) => <Legend {...args} />

export const GameLegend = Template.bind({})
