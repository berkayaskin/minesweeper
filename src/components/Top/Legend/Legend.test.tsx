import { render } from '@testing-library/react'
import { Legend } from './Legend'

it('should render legend correctly', () => {
  const { asFragment } = render(<Legend feature='Flag' firstAction='ctrl' secondAction='click' />)
  expect(asFragment()).toMatchSnapshot()
})
