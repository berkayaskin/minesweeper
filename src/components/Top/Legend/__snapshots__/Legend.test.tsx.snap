// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render legend correctly 1`] = `
<DocumentFragment>
  <code
    class="css-g5hezf"
  >
    <div
      class="chakra-stack css-14lxv93"
    >
      <h1
        class="chakra-text css-1adld2q"
      >
        Flag:
      </h1>
      <span
        class="chakra-text css-91rsr4"
      >
        ctrl
      </span>
      <p
        class="chakra-text css-1x3w0s1"
      >
        +
      </p>
      <span
        class="chakra-text css-1bjr0e8"
      >
        click
      </span>
    </div>
  </code>
</DocumentFragment>
`;
